app/src/main/kotlin/com/mapzen/erasermap/view/DynamicChildHeightRelativeLayout.kt

```diff
@@ -0,0 +1,44 @@
+package com.mapzen.erasermap.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.widget.RelativeLayout
+import java.util.ArrayList
+
+abstract class DynamicChildHeightRelativeLayout : RelativeLayout {
+
+  constructor(context: Context) : super(context) {}
+  constructor(context: Context, attributes: AttributeSet) : super(context, attributes) {}
+  constructor(context: Context, attributes: AttributeSet, style: Int) : super(context, attributes,
+      style) {}
+
+  abstract fun idsForDynamicChildren(): ArrayList<Int>
+
+  var views: ArrayList<View>? = null
+
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    views = ArrayList<View>()
+    for (i in 0..idsForDynamicChildren().size - 1) {
+      val viewId = idsForDynamicChildren()[i]
+      (views as ArrayList<View>).add(findViewById(viewId))
+    }
+  }
+
+  override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {
+    super.onLayout(changed, l, t, r, b)
```

Comment:
All I can think of is using reflection or copying the entirety of `RelativeLayout` into this class. I think I'll leave it as is for now and keep an eye out as you suggest